Firmware/ultralcd.cpp

```diff
@@ -902,9 +902,24 @@ void lcd_commands()
         switch (lcd_commands_step)
         {
         case 0:
-            lcd_commands_step = 3;
+            lcd_commands_step = 6;
             [[fallthrough]];
 
+        case 6:
+            enquecommand_P(G28W);
```

Comment:
I don't think this many steps are needed. Steps 4, 5 and 6 can be condensed into 3 enquecommand_P() calls for:
```
G28...
G1...
M400

```
This is possible since the cmdqueue should be big enough for all these gcodes to be enqueued at once.

Reply:
Combining G28W and G1... saves some bytes but the `st_synchronize()` in the same case doesn't work as the TM cal will stop as it isn't IDLE.